packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -232,14 +233,8 @@ mixin class ChangeNotifier implements Listenable {
   /// so that the method is tree-shaken away when the flag is false.
   @protected
   static void maybeDispatchObjectCreation(ChangeNotifier object) {
-    // Tree shaker does not include this method and the class MemoryAllocations
-    // if kFlutterMemoryAllocationsEnabled is false.
-    if (kFlutterMemoryAllocationsEnabled && !object._creationDispatched) {
-      FlutterMemoryAllocations.instance.dispatchObjectCreated(
-        library: _flutterFoundationLibrary,
-        className: '$ChangeNotifier',
-        object: object,
-      );
+    if (!object._creationDispatched) {
+      assert(debugMaybeDispatchCreated('foundation', 'ChangeNotifier', object));
       object._creationDispatched = true;
```

Comment:
Should this whole `if` be inside an assert? Is there any point in tracking `_creationDispatched` outside of debug builds? (Also, if we do that, we should rename that field to `_debugCreationDispatched`.